Toggl.Storage.Realm/PushRequestIdentifierStorage.cs

```
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Linq;
+using Toggl.Shared;
+using Toggl.Storage.Models;
+using Toggl.Storage.Realm.Models;
+using RealmDb = Realms.Realm;
+
+namespace Toggl.Storage.Realm
+{
+    internal sealed class PushRequestIdentifierStorage : IPushRequestIdentifierRepository
+    {
+        private const long onlyId = 1;
+
+        private readonly object storageAccess = new object();
+        private readonly IRealmAdapter<IDatabasePushRequestIdentifier> realmAdapter;
+
+        public PushRequestIdentifierStorage(IRealmAdapter<IDatabasePushRequestIdentifier> realmAdapter)
+        {
+            Ensure.Argument.IsNotNull(realmAdapter, nameof(realmAdapter));
+
+            this.realmAdapter = realmAdapter;
+        }
+
+        public void Clear()
+        {
+            lock (storageAccess)
+            {
+                try
+                {
+                    var record = realmAdapter.GetAll().FirstOrDefault();
+                    if (record == null)
+                        return;
+
+                    realmAdapter.Delete(onlyId);
+                }
+                catch
+                {
+                }
```

What is the reason for this empty catch? Shouldn't we at least log the error?

---

We could log the error, but it's gonna be somewhere on somebody's device and we won't have it anyway... at least with current setup. So I don't think it's an issue for us. But considering that it can only fail if IO derps (as it only calls into realm db), maybe we should actually let it crash the app (as per mobile guidelines) so that we get the error and the stack trace on appcenter.

any comments on this @heytherewill or @simonrozsival?

---

Delete should not throw anyway: https://realm.io/docs/dotnet/3.0.0/api/reference/Realms.Realm.html#Realms_Realm_Remove_Realms_RealmObject